src/components/Paginate.tsx

```diff
@@ -0,0 +1,213 @@
+import { chevronBackSharp, chevronForwardSharp } from "ionicons/icons";
+import { IonIcon } from "@ionic/react";
+import React, { FunctionComponent, useEffect, useState } from "react";
+
+const getPageCount = (
+  numberOfResults: number,
+  numberOfResultsPerPage: number
+): number => Math.ceil(numberOfResults / numberOfResultsPerPage);
+
+interface PaginateProps {
+  children: FunctionComponent;
+  limit: number;
+  onPageClick: (page: number) => void;
+  page: number;
+  totalResults: number;
+}
+
+export default function Paginate({
+  children,
+  limit,
+  onPageClick,
+  page,
+  totalResults,
+}: PaginateProps) {
+  const [currentPage, setCurrentPage] = useState(page);
+  const numberOfPages: number = getPageCount(totalResults, limit);
+
+  const displayLastResultsNumberProperly = (): string => {
```

Comment:
Any ideas for shorter names here?

Reply:
I think we can call it `displayLastResultsNumber` instead of `displayLastResultsNumberProperly`, since it's kind of redundant to say "properly", because we should always be displaying the last results number properly.

I think I added "properly" to remind whoever ends up reading this code in the future that it's very easily possible to display the last number incorrectly.

4b180d6d591e3dfd762c7f7fa1d34ef087fd2663 changes the function's name from [`displayLastResultsNumberProperly`](https://github.com/amilajack/tide/pull/290/commits/4b180d6d591e3dfd762c7f7fa1d34ef087fd2663#diff-ffa25bcbf841352bd24c796ad6d64a1ba743ea8290c4a9131ca833ad03cafa27L28) to [`displayLastResultsNumber`](https://github.com/amilajack/tide/pull/290/commits/4b180d6d591e3dfd762c7f7fa1d34ef087fd2663#diff-ffa25bcbf841352bd24c796ad6d64a1ba743ea8290c4a9131ca833ad03cafa27R28).